src/core/socialprovider.js

```diff
@@ -314,6 +314,7 @@ XMPPSocialProvider.prototype.sendMessage = function(to, msg, continuation) {
  * @private
  */
 XMPPSocialProvider.prototype.onMessage = function(msg) {
+  this.lastMessageDate_ = new Date();
```

Comment:
Can you add this to the constructor, just so we have a list of all internal variables there?